src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/service/TelemetryService.kt

```diff
@@ -19,17 +19,30 @@ class TelemetryService(
   }
 
   fun trackGoalCreateEvent(goal: Goal) {
-    telemetryClient.trackEvent(GOAL_CREATE_EVENT, goal.goalCreateEventCustomDimensions())
-    with(Span.current()) {
-      goal.goalCreateEventCustomDimensions().forEach { (key, value) -> setAttribute(key, value) }
-      addEvent("$GOAL_CREATE_EVENT-via-span")
-    }
+    telemetryClient.trackEvent(
+      GOAL_CREATE_EVENT,
+      getExistingCustomDimensions().plus(createEventCustomDimensions(goal)),
+    )
   }
 
-  private fun Goal.goalCreateEventCustomDimensions(): Map<String, String> =
-    mapOf(
-      "status" to this.status.name,
-      "stepCount" to this.steps.size.toString(),
-      "reference" to this.reference.toString(),
-    )
+  /**
```

Comment:
I've renamed this method because we didnt like it, and also changed it from being an ext. function because I think that might have been adding a bit of confusion.
The idea is that it is a method that returns the custom dimensions for a `goal-create` event (hence it's original name).

Later on I anticipate us having similar methods that create the custom dimensions for other events we might want to send - update Goal, add Step, delete Step, complete Goal, update goal review date .... etc etc ....